README.md
# Floppy Glory

This is a collection of DOS games found on various floppy disks from Allen High School's computer science department. DOSBox is required to run these games.

TODO:
* Attribute credit to creators where possible
* Fix some games that return errors of -3 and -4
* Find matching source code to games that only have binaries


QUEST_FOR_KNOWLEDGE/HISTORY.TXT
Scholars believe Egyptions built pyramids as:
Elaborate tombs for their pharaohs.
McDonalds
Interdiminsional portals
What? The aliens build them ofcourse.

Early hominids appeared as early as:
4,000,000 B.C.
Yesterday
100,000,000 B.C.
1,000,000 B.C.

Early hominids first appeared in:
Africa
India
Mesopotamia
Europe

The Paleolithic Age began in:
2,500,000 B.C.
1,600,000 B.C.
1,000,000 B.C.
100,000 B.C.

Homo Erectus first appeared when?
1,600,000 B.C.
1,000,000 B.C.
2,500,000 B.C.
100,000 A.D.

Neanderthals first appeared when?
200,000 B.C.
100,000 B.C.
1,000,000 B.C.
50,000 B.C.

Cro-Magnons first appeared when?
40,000 B.C.
30,000 B.C.
20,000 B.C.
10,000 B.C.

The last Ice Age ended when?
8,000 B.C.
10,000 B.C.
7,000 B.C.
20,000 B.C.

The Neolithic Age began when?
8,000 B.C.
7,000 B.C.
6,000 B.C.
10,000 B.C.

The Bronze Age began when?
3,000 B.C.
3,500 B.C.
5,000 B.C.
1,000 B.C.

The first civilization emerged in:
Sumer
America
Mars
Egypt

Upper and Lower Egypt united in:
3,100 B.C.
4,000 B.C.
2,000 B.C.
Upper and Lower Egypt never united

The Great Pyramid was build in:
2,600 B.C.
2,500 B.C.
2,000 B.C.
3,000 B.C.

The Xia Dynasty of China emerged in:
2,000 B.C.
1,000 B.C.
3,000 B.C.
0 A.D.

Hammurabi is famous for what achievement?
Codification of Laws
Discovery of Iron
Unification of Mesopotamia
First known painting

The Shang Dynasty of China began when?
1532 B.C.
1492 B.C.
1600 B.C.
1878 B.C.

The Zhou Dynasty of China began when?
1027 B.C.
987 B.C.
368 B.C.
238 B.C.

The Zhou Dynasty of Chine ended when?
256 B.C.
234 A.D.
398 B.C.
398 A.D.

The Upanishads were composed by the Hindus around:
750-550 B.C.
1000-750 B.C.
550-300 B.C.
100 B.C. - 100 A.D.

Assyrians conquer Israel around:
722 B.C.
789 B.C.
234 B.C.
234 A.D.

Buddha died around:
483 B.C.
420 B.C.
400 B.C.
200 A.D.

The first Nubian Empire, Kerma, arose when?
2,000 B.C.
1,500 B.C.
1,000 B.C.
500 B.C.

Hatshepsut became pharoah of Egypt in:
1472 B.C.
1400 B.C.
1378 B.C.
1262 B.C.

Confucious died in:
479 B.C.
459 B.C.
469 B.C.
429 B.C.

Confucious was from what area?
China
India
Africa
Mesopotamia

The Han Dynasty repleaced the Qin Dynasty in China when?
202 B.C.
158 B.C.
235 B.C.
636 B.C.

Xerxes was famous for what empire?
Persia
Greece
Assyria
China

Darius was famous for what empire?
Persia
Greece
Assyria
China

Shi Huangdi was famous for what empire?
China
Japan
Korea
Vietnam

Taharqa was famous for what empire?
Nubia
Egypt
Sumer
Assyria

Ashurbanipal was famous for what empire?
Assyria
Egypt
China
Nubia

Hatshepsut's stepson was who?
Thutmose III
Thutmose II
Ashurbanipal
Alexander

Napata was the capitol of what empire?
Kush
Assyria
Sumer
Persia

Piankhi was the king of what empire?
Kush
Assyria
Sumer
Persia

Secnnacherib was king of what empire?
Assyria
Kush
Sumer
Persia

After being conqered by the Assyrians, what kindom rose in Babylon
Chaldean
Persian
Assyrian
Egyptian

Nebuchadnezzar was king of what empire?
Chaldea
Persia
Kush
Sumer

Cyrus was famous for what empire?
Persia
Assyria
Kush
Sumer

Cambyses was the son of what king of Persia?
Cyrus
Darius
Xerxes
Darius II

Minoan civilization thrived in Crete in which millinia?
2000 B.C.
3000 B.C.
1000 B.C.
4000 B.C.

Mycenaeans invaded Crete in:
1450 B.C.
1350 B.C.
1250 A.D.
2000 B.C.

The Trojan war took place in:
1200 B.C.
1150 B.C.
1250 B.C.
1000 B.C.

Homer composed his epics in:
800 B.C.
700 B.C.
600 B.C.
500 B.C.

Philip conqered Greece in:
338 B.C.
320 B.C.
300 B.C.
275 B.C.

Alexander began to build his empire in:
334 B.C.
320 B.C.
300 B.C.
275 B.C.

Alexander was the son of what conqerer?
Philip
Conan
Caesar
Maximus

Roman laws are carved on the Twelve Tables in:
451 B.C.
420 B.C.
300 B.C.
280 B.C.

The Punic wars began in:
264 B.C.
300 B.C.
200 B.C.
200 A.D.

The Punic wars were a conflict between Rome and...
Carthage
Persia
Greece
Assyria

Rome sacked Carthage in:
146 B.C.
120 B.C.
53 B.C.
232 A.D.

Julius Caesar was killed in...
44 B.C.
40 B.C.
30 B.C.
52 B.C.

Octavian became emperor of Rome in...
27 B.C.
22 B.C.
38 B.C.
41 B.C.

The 400 year period of peace in Rome was known as...
Pax Romona
E Pluribus Unum
Pax Infinitus
Romana Infinitus

Pax Romana began in...
27 B.C.
23 A.D.
27 A.D.
39 B.C.

Rome divided itself into Eastern and Western empires in...
285 A.D.
200 A.D.
355 A.D.
450 A.D.

Constantinople was establised as the new capitol of Rome in...
330 A.D.
300 A.D.
255 A.D.
400 A.D.

The Western Roman empire fell in...
476 A.D.
476 B.C.
514 A.D.
514 B.C.

The Olmec culture of Mexico rose in...
1200 B.C.
1000 B.C.
800 B.C.
800 A.D.

Hannibal was a general for what Empire?
Carthage
Rome
Assyria
Persia

Scipio was a general for what Empire?
Rome
Carthage
Assyria
Persia

Constantine converted to Christianity in...
312 A.D.
298 A.D.
329 B.C.
456 A.D.

Virgil is famous for what work of literature?
The Aeneid
The Oddyssey
The Bible
The first dictionary

Homer is famous for what work of literature?
The Oddyssey
The Aeneid
The Bible
The first dictionary

Greek fire was invented in what city?
Constantinople
Athens
Sparta
Troy

The Eastern Roman Empire assumed the name of...
Byzantine
Croatia
Serbia
Turkey

The Christian church divided for the first time in...
1054 A.D.
1022 B.C.
978 A.D.
1054 B.C.

Constantinople fell to the Turks in...
1453
1293
1893
1287

Gunpowder was invented in China in...
850
328
873 B.C.
239 B.C.

Genghis Khan begins his Mongolian conquest in...
1209
987
1102
1054

Charles Martel defeated what religious group at the Battle of Tours?
Muslims
Jews
Hindus
Buddhists

When did the Battle of Tours take place?
732
700
863
632

MAX/AAA.CPP
// Andrew Closson, Mahmood Subhani
// Period 6
// Mr. Baker
// Program: Max's Escape


#include<Iostream.h>
#include<conio.h>n
#include<apstring.h>
#include<iomanip.h>
#include<apvector.h>
#include<graphics.h>
#include<stdlib.h>
#include<dos.h>
#include<fstream.h>
#include<alloc.h>
//constants



//structure

struct monster      //struct for monster info
{

int X, Y;           //monster coordinates
int State;          //monster movement type
int Timer;          //speed of movement

};



//variables
	int grdriver;     //graphics function
	int grmode;       //graphics function
	int errorcode;    //graphics function
	int X;            //matrix "X" axis
	int Y;            //matrix "Y" axis
	int I;            //LCV used to display graphics
	int J;            //another LCV used to display graphics
	int Debug;
	int Havekey;      //shows if Max has the key
	int Choice;       //option on main menu

	int Playerx, Playery;   //Max's coordinates

	int Level;              //used to load level
	int Exit;               //used to exit when needed
	int Map[30][30];        //matrix used for maps

	apvector <monster> Badie(30);  //stores info on all monsters

	ifstream file;                 //used for files (maps and images)
	unsigned Size;                      //used for size of tiles
	apvector <char far *> maplist(15);  //used for loading maps
	apvector <int> points(768);         //used to plot all of the point for images


//prototypes
	void initialize();
	void changemap();
	void drawscreen();
	void updatescreen();
	void movement();
	void think();
	void show ();
	void maketile ();
	void title ();
	void lose();
	void win();


	//Tiles

	void far *Wall;
	void far *Door;
	void far *Key;
	void far *Max;
	void far *Monster;
	void far *Space;


void main()
{

Debug = 1;


initialize();
title();
cleardevice();
changemap();

do{

updatescreen();
movement();
think();

}while(!Exit);



closegraph();
}




void initialize()
{

int Loop;


	show();

		setbkcolor(0);
		setcolor(15);

		//Load maplist file

		file.open("maplist.lst");

		for(Loop = 0; Loop < 15; Loop++)
		{
		file>>maplist[Loop];
		}
		file.close();


		//Load Tiles and store to memory

		file.open("A:\\Wall.icf");

		maketile();

		Size=imagesize(0,0,15,15);

		Wall=farmalloc(Size);

		getimage(0,0,15,15,Wall);

		file.close();
		cleardevice();




		file.open("A:\\Key.icf");

		maketile();

		Size=imagesize(0,0,15,15);

		Key=farmalloc(Size);

		getimage(0,0,15,15,Key);

		file.close();
		cleardevice();



		file.open("A:\\Max.icf");

		maketile();

		Size=imagesize(0,0,15,15);

		Max=farmalloc(Size);

		getimage(0,0,15,15,Max);

		file.close();
		cleardevice();




		file.open("A:\\Monster.icf");

		maketile();

		Size=imagesize(0,0,15,15);

		Monster=farmalloc(Size);

		getimage(0,0,15,15,Monster);

		file.close();
		cleardevice();



		file.open("A:\\Door.icf");

		maketile();

		Size=imagesize(0,0,15,15);

		Door=farmalloc(Size);

		getimage(0,0,15,15,Door);

		file.close();
		cleardevice();




		Size=imagesize(0,0,15,15);

		Space=farmalloc(Size);

		getimage(0,0,15,15,Space);

		cleardevice();
};


void changemap()                  //Called at the end of every level
{
ifstream Mapfile;
//char far * Load;

int Monsternum = 0;



//Load = maplist[Level];

switch(Level)
	{
	case 0: Mapfile.open("a:\\map1.map"); break;
	case 1: Mapfile.open("a:\\map2.map"); break;
	case 2: Mapfile.open("a:\\map3.map"); break;
	case 3: Mapfile.open("a:\\map4.map"); break;
	case 4: Mapfile.open("a:\\map5.map"); break;
	case 5: Mapfile.open("a:\\map6.map"); break;
	case 6: Mapfile.open("a:\\map7.map"); break;
	case 7: Mapfile.open("a:\\map8.map"); break;
	case 8: Mapfile.open("a:\\map9.map"); break;
	case 9: Mapfile.open("a:\\map10.map"); break;
	case 10: win(); break;
	}




for(Y = 0; Y < 25; Y++)
	for(X = 0; X < 25; X++)
		{
		Mapfile>>Map[X][Y];

		if(Map[X][Y] == 5)            //Spawn Monster
			{
			Badie[Monsternum].X = X;
			Badie[Monsternum].Y = Y;
			Badie[Monsternum].State = 1;
			Monsternum++;
			}

		if(Map[X][Y] == 4)            //Find Player Start
			{
			Playerx = X;
			Playery = Y;

		//		if(Debug)
		//			cout << "Found Player Start." << endl;
			}
		}

Mapfile.close();
Level++;

//if(Debug)
//	cout << "End changemap()" << endl;


drawscreen();

};


void drawscreen()                 //Draws all the tiles
{

int Xoffset, Yoffset;

Xoffset = Yoffset = 0;



	for(Y=0;Y<25;Y++)
	{
		for(X=0;X<25;X++)
		{

			switch(Map[X][Y])
			{

				case 0: putimage(Xoffset, Yoffset, Space, COPY_PUT); break;
				case 1: putimage(Xoffset, Yoffset, Wall, COPY_PUT); break;
				case 2: putimage(Xoffset, Yoffset, Door, COPY_PUT); break;
				case 3: putimage(Xoffset, Yoffset, Key, COPY_PUT); break;
				case 4: putimage(Xoffset, Yoffset, Max, COPY_PUT); break;
				case 5: putimage(Xoffset, Yoffset, Monster, COPY_PUT); break;
			}
		Xoffset+=17;
		}
	Yoffset+=17;
	Xoffset = 0;
	}

};


void updatescreen()
{

int Xoffset, Yoffset;

Xoffset = Yoffset = 0;



	for(Y=0;Y<25;Y++)
	{
		for(X=0;X<25;X++)
		{

			switch(Map[X][Y])
			{

				case 0: putimage(Xoffset, Yoffset, Space, COPY_PUT); break;
				case 1: break;
				case 2: break;
				case 3: break;
				case 4: putimage(Xoffset, Yoffset, Max, COPY_PUT); break;
				case 5: putimage(Xoffset, Yoffset, Monster, COPY_PUT); break;
			}
		Xoffset+=17;
		}
	Yoffset+=17;
	Xoffset = 0;
	}


}


void movement()                    //Checks for input and moves player
{
int Keyhit;
int Tile;

int Oldpx, Oldpy;

Oldpx = Playerx;
Oldpy = Playery;

Map[Playerx][Playery] = 0;


if(kbhit())
	Keyhit = getch();
else
	Keyhit = 0;


switch(Keyhit)
	{
	case 72: Playery--; break;

	case 80: Playery++; break;

	case 75: Playerx--; break;

	case 77: Playerx++; break;

	case 27: Exit = 1; break;
	}


Tile = Map[Playerx][Playery];


switch(Tile)                                          //Collision Detection
	{
	case 0: Map[Playerx][Playery] = 4; break;
	case 1: Playerx = Oldpx; Playery = Oldpy; break;
	case 2: if(Havekey == 1)
						Map[Playerx][Playery] = 4;
					else
						{
						Playerx = Oldpx;
						Playery = Oldpy;
						}
						break;
	case 3: Map[Playerx][Playery] = 4; Havekey = 1; break;
	case 4: break;
	case 5: Map[Playerx][Playery] = 5; lose(); break;
	case 6: changemap(); break;
	}
};



void think()
{
int Loop;
int Dir;
int Moved;

randomize();

for(Loop = 0; Loop < 30; Loop++)
	{

	if(Badie[Loop].State != 0)
		Map[Badie[Loop].X][Badie[Loop].Y] = 0;


	if(Badie[Loop].Timer < 3)
		{
			if(Badie[Loop].State == 1)               //Random Movement
				{

					Dir = random(4);



					if(Dir == 1)
						if(Map[Badie[Loop].X + 1][Badie[Loop].Y] == 0)
							{
							Badie[Loop].X++;
							//Moved = 1;
							}
					if(Dir == 2)
						if(Map[Badie[Loop].X - 1][Badie[Loop].Y] == 0)
							{
							Badie[Loop].X--;
							//Moved = 1;
							}
					if(Dir == 3)
						if(Map[Badie[Loop].X][Badie[Loop].Y + 1] == 0)
							{
							Badie[Loop].Y++;
							//Moved = 1;
							}
					if(Dir == 4)
						if(Map[Badie[Loop].X][Badie[Loop].Y - 1] == 0)
							{
							Badie[Loop].Y--;
							//Moved = 1;
							}

				//if(Map[Badie[Loop].X][Badie[Loop].Y] == 4);
					//lose();
				}

		}
	Map[Badie[Loop].X][Badie[Loop].Y] = 5;

	if(Badie[Loop].Timer < 5)
		Badie[Loop].Timer++;
	else
		Badie[Loop].Timer=0;
	}
};


void show()

{
	grdriver=VGA;
	grmode=VGAHI;
	initgraph(&grdriver, &grmode, "c:\\TC\\BGI");
	errorcode=graphresult();

	if(errorcode!=grOk)
	{clrscr();
	cout<<"Error";
	getch();
	exit(1);
}

};


void maketile()                       //Deal with graphic file
{
for(I=0;I<768;I++)
		{
			file>>points[I];
		}

	I=0;

	for(J=0;J<256;J++)
		{
			putpixel(points[I],points[I+1],points[I+2]);
			I=I+3;
		}
}


void title()
{
int Choice;
int midx;

cleardevice();


 midx = getmaxx() / 2;

settextstyle(3, HORIZ_DIR, 1);
setcolor(3);
outtextxy(midx-185, 120, "By: Andrew Closson and Mahmood Subhani");
settextstyle(2, HORIZ_DIR, 4);
setcolor(4);
outtextxy(midx-60, 150, "Please type a number:");
outtextxy(midx-40, 200, "1) Play");
outtextxy(midx-40, 220, "2) Instructions");
outtextxy(midx-40, 240, "3) Quit");

I=1;
do{
setcolor(I);
settextstyle(4, HORIZ_DIR, 6);
outtextxy(midx-75, 3, "Max's");
outtextxy(midx-85, 50, "Escape");

setcolor(0);
outtextxy(midx-75, 3, "Max's");
outtextxy(midx-85, 50, "Escape");
I++;
}while(!kbhit());



Choice=getch();

switch (Choice)
{

	case 49: break;


	case 50: cleardevice();
					 setcolor(4);
					 settextstyle(1, HORIZ_DIR, 1);
					 outtextxy(midx-180, 3, "Use the arrow keys to move Max.");
					 outtextxy(midx-310, 50, "Press Esc during game play when you want to return to the menu");
					 getch();
					 cleardevice();
					 title();            //Note: Recursive function call
					 break;



	case 51: Exit=1; break;

}

};


void lose()
{
int Choice;
int I;



cleardevice();
setcolor(3);
settextstyle(1, HORIZ_DIR, 1);
outtextxy(5, 3, "The monster drags your corpse off to a dark corner...");
outtextxy(5, 100, "Sorry, you lost.  Please play again.");
delay(140);
getch();
cleardevice();


Exit = 1;

};

void win()
{
int Choice;



cleardevice();
setcolor(4);
settextstyle(4, HORIZ_DIR, 3);
outtextxy(5, 3, "Congratulations,");
outtextxy(5, 100, "Max got out of the dungeon and flew back to Earth");
getch();

setcolor(1);
settextstyle(1, HORIZ_DIR, 1);
outtextxy(5, 300, "Please play again");
getch();

Exit = 1;

};